src/main/java/org/threadly/concurrent/UnfairExecutor.java

```diff
@@ -0,0 +1,267 @@
+ * scheduler does not ensure that tasks are executed in the order they are submitted, but rather 
+ * tasks are consumed however is fastest.</p>
+ * 
+ * <p>This executor has an execution queue per thread.  That way each thread has a many producer, 
+ * one consumer safety guarantees.  Vs normal pools which have to deal with a many to many thread 
+ * safety issue.  Determining the thread queue may be based on a variety of information, some 
+ * examples are the {@link Object#hashCode()} of the provided task, the current time, or other 
+ * weak choices to try and distribute work evenly.</p>
+ * 
+ * <p>In this it is obviously possible for one long running task to block other tasks needing to 
+ * run (even if other threads are idle).  Because of that tasks should be equally sized, having 
+ * large tasks mixed in with small ones will make this more problematic.  We also recommend having 
+ * thread counts which are prime numbers for a more even hash distribution.</p>
+ * 
+ * @author jent - Mike Jensen
+ * @since 4.5.0
+ */
+public class UnfairExecutor extends AbstractSubmitterExecutor {
+  private final Worker[] schedulers;
+  private final AtomicBoolean shutdownStarted;
+  
+  /**
+   * Constructs a new {@link UnfairExecutor} with a provided thread count.  This defaults to using 
+   * daemon threads.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   */
+  public UnfairExecutor(int threadCount) {
+    this(threadCount, true);
+  }
+
+  /**
+   * Constructs a new {@link UnfairExecutor} with a provided thread count.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   * @param useDaemonThreads {@code true} if created threads should be daemon
+   */
+  public UnfairExecutor(int threadCount, boolean useDaemonThreads) {
+    this(threadCount, new ConfigurableThreadFactory(UnfairExecutor.class.getSimpleName() + "-", true, 
+                                                    useDaemonThreads, Thread.NORM_PRIORITY, null, null));
+  }
+
+  /**
+   * Constructs a new {@link UnfairExecutor} with a provided thread count and factory.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   * @param threadFactory thread factory for producing new threads within executor
+   */
+  public UnfairExecutor(int threadCount, ThreadFactory threadFactory) {
+    ArgumentVerifier.assertGreaterThanZero(threadCount, "threadCount");
+    
+    schedulers = new Worker[threadCount];
+    shutdownStarted = new AtomicBoolean(false);
+    for (int i = 0; i < threadCount; i++) {
+      schedulers[i] = new Worker(threadFactory);
+      schedulers[i].start();
+    }
+  }
+  
+  @Override
+  protected void doExecute(Runnable task) {
+    if (shutdownStarted.get()) {
+      throw new RejectedExecutionException("Pool is shutdown");
+    }
+    
+    long seed = Math.abs(task.hashCode() ^ Clock.lastKnownTimeNanos());
```

Comment:
Is there a reason you dont want negative here?  seems like it would just 1/2 your hash space but not cause any other problems then that.